src/ImageSegmentation.jl

```diff
@@ -5,6 +5,7 @@ import Base: show
 using LinearAlgebra, Statistics
 using DataStructures, StaticArrays, ImageCore, ImageFiltering, ImageMorphology, LightGraphs, SimpleWeightedGraphs, RegionTrees, Distances, StaticArrays, Clustering, MetaGraphs
 using ImageCore.ColorVectorSpace: MathTypes
+using ImageBase.ImageCore: GenericGrayImage, GenericImage
```

Comment:
We should either remove `ImageCore` from `Project.toml` or use

```suggestion
using ImageCore: GenericGrayImage, GenericImage
```